### ISSUE TITLE
"User User Root" and "User root slot" or extremely similar

### ISSUE THREAD

**Issue by epicEaston197** [Yellow-Dog-Man/Resonite-Issues#2350]
### Is your feature request related to a problem? Please describe.

These two nodes have a really similar name and could easily be mistaken from one another the output is also the exact same color to the human eye which is really annoying when working with user root operations
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/76523692/b48d077c-d8ed-4beb-a8fd-bbab70de68b9)

### Describe the solution you'd like

I would like either a name change or for the output of the User user root node to have a different color because the color of that output looks exactly like a slot type when it isn't

### Describe alternatives you've considered

None

### Requesters

@epicEaston197

**Comment by JackTheFoxOtter** [comment]
I think changing the color of the UserRoot type would help best here, it's a small tweak but I think it makes sense. Having any type look visually similar to the Slot type is quite confusing.

**Comment by shiftyscales** [comment]
If I recall, most type colours in Resonite are procedurally generated rather than explicitly defined- so it's expected there will be some possible overlap / clash. I believe the main exceptions to these are certain value types, e.g. bools being black, floats being blue, ints being green, etc.

There have been discussions, e.g. in #1947 to add additional distinguishing features like type markers to better / more easily identify certain types even when colours are near identical.

**Comment by Banane9** [comment]
> If I recall, most type colours in Resonite are procedurally generated rather than explicitly defined- so it's expected there will be some possible overlap / clash. I believe the main exceptions to these are certain value types, e.g. bools being black, floats being blue, ints being green, etc.

`Slot`, `User` and some other basic types are explicitly defined as well - `UserRoot` happening to have an almost identical color to `Slot` is unfortunate, but could be fixed.

**Comment by Frooxius** [comment]
Yes, they are procedural. We can assign this an explicit color too to fix it as well.

Generally I do recommend not relying on color too much - there's a lot of types and only so many distinctly looking colors. 

Question is, what color would be suitable here?

**Comment by AmasterAmaster** [comment]
Same hue (same family of nodes that it relates to), different saturation/contrast (either brighter or darker). That is my suggestion.

**Comment by Banane9** [comment]
> Question is, what color would be suitable here?

I would also say something in the direction of the User color, maybe darker because it's the "base" / root?

**Comment by JackTheFoxOtter** [comment]
Yeah, a dark purple perhaps. I agree that you can't have every type have a unique color, but I'd say for something as common as Slot and UserRoot, they should be distinguishable.

**Comment by Psychpsyo** [comment]
Two things having very similar colors is fine most of the time, when they will almost never be used together but slots are used pretty much always, and often together with user roots so those should probably be distinct.
